### ISSUE TITLE
Transformed from Babel format to Esprima format AST doesn't handle function defaults

### ISSUE THREAD

**Issue by DmitrySoshnikov** [babel/babel-eslint#159]
For this code:

```js
function foo(x = 10) {}
```

the esprima AST for the function node should contain `defaults` property (in Babel they are in the `params` with the `AssignmentPattern` node type).

**Comment by DmitrySoshnikov** [comment]
Oh, seems like it was removed https://github.com/eslint/espree/commit/9e99a3593c668d43527ee9dfc3f6470e506d17c4, and conversions is made not into actual esprima AST format anymore.
